### PR TITLE
[Clang] Set `__cpp_explicit_this_parameter`

### LINKED ISSUE #82780: `__cpp_explicit_this_parameter` not set in clang-18
It seems that clang-18  [supports explicit object parameter (deducing this)](https://en.cppreference.com/w/cpp/compiler_support/23) however the feature test macro is not set: https://godbolt.org/z/j6za1evK9

### PR DESCRIPTION
**Pull request by cor3ntin**
There are not a lot of outstanding known issues
with deducing this (besides #95112), so it
seems reasonable to claim full support.

Fixes #82780

### CODE REVIEW

**Comment by erichkeane** [comment]
It appears that https://github.com/llvm/llvm-project/pull/95112 is actually pretty close... are we better off holding off on this until after that goes in?

**Comment by cor3ntin** [comment]
There is  no urgency. I made the pr so we don't forget (I certainly had...)

**Comment by cor3ntin** [comment]
@erichkeane ping

File: clang/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -272,6 +272,8 @@ C++23 Feature Support
 
 - Extend lifetime of temporaries in mem-default-init for P2718R0. Clang now fully
   supports `P2718R0 Lifetime extension in range-based for loops <https://wg21.link/P2718R0>`_.
+  
+- ``__cpp_explicit_this_parameter`` is now defined. (#GH82780)
```
Comment (erichkeane) [inline clang/docs/ReleaseNotes.rst:276]: As a nit, "is now defined to the value specified in the C++X specification." 

perhaps?

**Review by erichkeane — APPROVED** [review]

**Pull request merged**